ruby/app.rb

```diff
@@ -38,19 +40,33 @@
 end
 
 get '/transactions' do
-  now = Date.today
-  thirty_days_ago = (now - 30)
-  begin
-    transactions_response = client.transactions.get(access_token, thirty_days_ago, now)
-  rescue Plaid::ItemError => e
-    transactions_response = { error: {error_code: e.error_code, error_message: e.error_message}}
-  end
+  start_date = params[:start_date] || '2016-07-12'
+  end_date = params[:end_date] || '2017-01-09'
+
+  transactions = PlaidClient.get_transactions(access_token, start_date, end_date)
+  company_name_to_info = get_company_info(transactions)
+  sorted_transactions = sort_and_format_transactions(transactions, company_name_to_info)
   content_type :json
-  transactions_response.to_json
+  {transactions: sorted_transactions}.to_json
 end
 
 get '/create_public_token' do
   public_token_response = client.item.public_token.exchange(access_token)
   content_type :json
   public_token_response.to_json
 end
+
+def sort_and_format_transactions(transactions, company_name_to_info)
+  transactions.sort_by(&:date).reverse.map do |transaction|
+    transaction.to_h.reduce({}) {|r,(k,v)| r[k.to_s] = v; r}.merge(company_name_to_info[transaction.name])
```

Comment:
```
transaction.to_h.reduce({}) {|r,(k,v)| r[k.to_s] = v; r}
``` 
added due to tests using OpenStruct, which symbolizes keys instead of stringifying them upon `to_h`